### ISSUE TITLE
GRPC Wrong Number of arguments

### ISSUE THREAD

**Issue by FeminismIsAwesome** [newrelic/newrelic-ruby-agent#1337]
When we upgraded the newrelic gem and deployed, ran into an issue with it seeming to fail when trying to prepend to GRPC when integrating with a google GRPC gem.

## Description
<img width="1179" alt="Screen Shot 2022-08-22 at 1 13 59 PM" src="https://user-images.githubusercontent.com/5641692/186010537-4b8d0dbc-3b9b-496f-b8bb-86f7dc0bc88c.png">

<img width="1095" alt="Screen Shot 2022-08-22 at 1 10 56 PM" src="https://user-images.githubusercontent.com/5641692/186010429-a5831d9b-d626-44eb-b131-d99f118a4b3d.png">

Using newrelic 8.9.0 and  google-ads-googleads 18.0.0 and grpc 1.48.0

## Expected Behavior
For it not to blow up by upgrading newrelic by default

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results
Provide any other relevant log data.
TIP:  Scrub logs and diagnostic information for sensitive information 

## Steps to Reproduce
```
# this is one of the ways specified in https://github.com/googleads/google-ads-ruby
Google::Ads::GoogleAds::GoogleAdsClient.new('config/google_ads_config.rb').service.google_ads
```
Tried auto, prepend, and chain arguments for grpc_client. only got it to work once i disabled it.

## Your Environment
ruby: 3.1.2
Using newrelic 8.9.0 and  google-ads-googleads 18.0.0 and grpc 1.48.0

## For Maintainers Only or Hero Triaging this bug
*Suggested Priority (P1,P2,P3,P4,P5):*
*Suggested T-Shirt size (S, M, L, XL, Unknown):*

**Comment by fallwith** [comment]
Hi @FeminismIsAwesome,

Thank you for bringing this issue to our attention. We are sorry to hear that upgrading to v8.10.0 caused this problem. We're very glad to see that you were able to disable the gRPC instrumentation to get things working while we figure out a fix.

In the issue title, you have the text "wrong number of arguments". Did that text originally come from a Ruby exception? If so, do you have the complete exception message and/or a Ruby backtrace? Your screenshots and version numbers for everything have already provided us with a great starting point to troubleshoot from, but any additional messages and traces you might have available to share would be much appreciated.

**Comment by FeminismIsAwesome** [comment]
Hey @fallwith ,

confirming this is what you want by stack trace? Here's a trace from my current rails console. I'll see if I can get a dummy app up that can reproduce the same error.

[1] pry(main)> Google::Ads::GoogleAds::GoogleAdsClient.new('config/google_ads_config.rb').service.google_ads
ArgumentError: wrong number of arguments (given 3, expected 2)
from /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/grpc-1.48.0-x86_64-darwin/src/ruby/lib/grpc/generic/client_stub.rb:98:in `initialize'
[2] pry(main)> wtf
Exception: ArgumentError: wrong number of arguments (given 3, expected 2)
--
0: /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/grpc-1.48.0-x86_64-darwin/src/ruby/lib/grpc/generic/client_stub.rb:98:in `initialize'
1: /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/newrelic_rpm-8.10.0/lib/new_relic/agent/instrumentation/grpc/client/prepend.rb:18:in `initialize'
2: /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/grpc-1.48.0-x86_64-darwin/src/ruby/lib/grpc/generic/service.rb:158:in `initialize'
3: /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/gapic-common-0.6.1/lib/gapic/grpc/service_stub.rb:63:in `new'
4: /Users/iannorris/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/gapic-common-0.6.1/lib/gapic/grpc/service_stub.rb:63:in `initialize'

**Comment by fallwith** [comment]
Thank you, @FeminismIsAwesome. This trace from Rails/Pry is perfect. We'll keep you posted on our progress with this one.

**Comment by FeminismIsAwesome** [comment]
https://github.com/FeminismIsAwesome/reproducing_newrelic_issue I was also able to make a Rails repo that was divorced from all the logic of our current app, in case that helps!
https://github.com/FeminismIsAwesome/reproducing_newrelic_issue/blob/main/spec/failing_spec.rb

**Comment by fallwith** [comment]
Thank you, @FeminismIsAwesome. This is very helpful indeed!

**Comment by fallwith** [comment]
Hi @FeminismIsAwesome,

With PR #1339 that was just merged into our dev branch, the New Relic Ruby agent will no longer make the `initialize` `super` call that would end up passing the wrong number of arguments when `google-ads-googleads` is involved.

Thank you again for bringing this issue to our attention, for the benefit of all Google Ads and gRPC users.

- The fix will be released with v8.11.0 of the New Relic Ruby agent gem, `newrelic_rpm`
- The fix is available via GitHub now for anyone who wants to set their `Gemfile` entry to `gem 'newrelic_rpm', git: 'https://github.com/newrelic/newrelic_ruby_agent'`
- The agent's gRPC client instrumentation can optionally be disabled entirely by setting the config file parameter `instrumentation.grpc_client` to a value of `disabled` or by setting the `NEW_RELIC_INSTRUMENTATION_GRPC_CLIENT` environment variable to `disabled`. More information on this can be found in the [gRPC instrumentation documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/instrumented-gems/grpc-instrumentation/#grpc-client-options)

**Comment by fallwith** [comment]
@FeminismIsAwesome @vitalinfo @sethboyles Version 8.10.1 of the `newrelic_rpm` gem has now been released and it contains a fix for the gRPC compatibility issues discussed here.
